alacritty_terminal/src/selection.rs

```diff
@@ -18,52 +18,53 @@
 //! finalized when the button is released. The selection should be cleared
 //! when text is added/removed/scrolled on the screen. The selection should
 //! also be cleared if the user clicks off of the selection.
-use std::cmp::{max, min};
+use std::mem::swap;
 use std::ops::Range;
 
 use crate::index::{Column, Point, Side};
 use crate::term::cell::Flags;
 use crate::term::{Search, Term};
 
-/// Describes a region of a 2-dimensional area
+/// Describes a type of a 2-dimensional area
 ///
-/// Used to track a text selection. There are three supported modes, each with its own constructor:
+/// Used to track a text selection type. There are three supported modes, each with its own constructor:
 /// [`simple`], [`semantic`], and [`lines`]. The [`simple`] mode precisely tracks which cells are
 /// selected without any expansion. [`semantic`] mode expands the initial selection to the nearest
 /// semantic escape char in either direction. [`lines`] will always select entire lines.
```

Comment:
> There are three supported modes, each with its own constructor

This comment part doesn't fit the enum. There should also not be any duplication between the different documentations, since that would mean both have to be edited on a change and people are likely to miss that.